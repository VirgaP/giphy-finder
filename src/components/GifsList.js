import React, {useState, useEffect} from 'react'
import { Button, Col, Row, Container } from 'react-bootstrap'
import 'bootstrap/dist/css/bootstrap.min.css';
import Modal from './MyModal'
import Gif from './Gif'

const GifsList = ({data, callback}) => {
    const [selected, setSelected] = useState({})
    const [open, setOpen] = useState(false)

    const handleClick=(item)=>{
        setSelected(item)
        setOpen(true)
    }

    useEffect(() => {
        window.addEventListener('scroll', handleScroll);
        return () => window.removeEventListener('scroll', handleScroll);
      }, []);
    
    const gifsList = data.map((item) =>
        <div className="gif-wrapper" key={item.id}>
            <Gif url={item.images.fixed_height_small.url} key={item.id} alt={item.slug} height={item.images.fixed_height_small.height} width={125} />
            <Button variant="light" onClick={()=>handleClick(item)}>show</Button>
        </div>
    )

    function handleScroll() {
        if (window.innerHeight + document.documentElement.scrollTop !== document.documentElement.offsetHeight) return;
        callback(true)
    }


    return (
        <>
        {!!open && <Modal item={selected} open={open} callback={setOpen}/>}
        {data.length > 0 && <div className="m-auto center">Scroll down to load more GIFs</div>}
            <div className="gifs-list" id="gifsList">{gifsList}</div> 
        </>
    )
}

export default GifsList
