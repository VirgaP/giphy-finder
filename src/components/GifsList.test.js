import React from 'react';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import GifsList from './GifsList';

configure({adapter: new Adapter()});

describe(GifsList, () => {
    const component = shallow(< GifsList data={[]}/>);
    it('should exist', () => {
        const component = renderer.create(<GifsList data={[]}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
