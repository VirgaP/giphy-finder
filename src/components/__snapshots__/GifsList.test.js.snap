// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifsList should exist 1`] = `
<div
  className="gifs-list m-auto"
  id="gifsList"
/>
`;
